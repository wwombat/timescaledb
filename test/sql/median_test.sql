-------------------------------------------
-- Basic even test, integer.
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_even"(val int);
INSERT INTO "mediantest_even" (val) VALUES
    (1),
    (3),
    (1),
    (3),
    (2);
-- ASCERTAIN
SELECT median(val) FROM mediantest_even;

-------------------------------------------
-- Basic odd test, integer.
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_odd"(val int);
INSERT INTO "mediantest_odd" VALUES
    (3),
    (1),
    (3),
    (2);
-- ASCERTAIN
SELECT median(val) FROM mediantest_odd;

-------------------------------------------
-- Repeated values.
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_repeated"(val int);
INSERT INTO "mediantest_repeated"
SELECT 3 FROM generate_series(0, 1000) as T(i);
-- ASCERTAIN
SELECT median(val) FROM mediantest_repeated;

-------------------------------------------
-- Float test.
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_float"(val float);
INSERT INTO "mediantest_float" VALUES
    (3.0),
    (1.3),
    (90000.0),
    (-3),
    (15.123141),
    (-134141);
-- ASCERTAIN
SELECT median(val) FROM mediantest_float;

-------------------------------------------
-- Empty table.
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_empty"(val float);
-- ASCERTAIN
SELECT median(val) FROM mediantest_empty;

-------------------------------------------
-- Gigantic odd table
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_gigantic_odd"(val int);
INSERT INTO mediantest_gigantic_odd (val)
SELECT i FROM generate_series(0, 100001) as T(i);
-- ASCERTAIN
SELECT median(val) FROM mediantest_gigantic_odd;

-------------------------------------------
-- Gigantic even table
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_gigantic_even"(val int);
INSERT INTO mediantest_gigantic_even (val)
SELECT i FROM generate_series(0, 100000) as T(i);
-- ASCERTAIN
SELECT median(val) FROM mediantest_gigantic_even;

-------------------------------------------
-- Multiple Selects
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_multiple"(a int, b int);
INSERT INTO mediantest_multiple (a, b)
SELECT i, 1000000 - i FROM generate_series(0, 10000) as T(i);
-- ASCERTAIN
SELECT median(a), median(b) FROM mediantest_multiple;

-------------------------------------------
-- Strings
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_text"(val text);
INSERT INTO mediantest_text(val)
SELECT i::text FROM generate_series(0, 100000) as T(i);
-- ASCERTAIN
SELECT median(val) FROM mediantest_text;

-------------------------------------------
-- Timestamp
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_date"(val timestamp);
INSERT INTO mediantest_date(val)
SELECT TIMESTAMP 'epoch' + (i * INTERVAL '1 second')
FROM generate_series(0, 100000) as T(i);
-- ASCERTAIN
SELECT median(val) FROM mediantest_date;
