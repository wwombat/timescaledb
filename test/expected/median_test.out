-------------------------------------------
-- Basic even test, integer.
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_even"(val int);
INSERT INTO "mediantest_even" (val) VALUES
    (1),
    (3),
    (1),
    (3),
    (2);
-- ASCERTAIN
SELECT median(val) FROM mediantest_even;
 median 
--------
      2
(1 row)

-------------------------------------------
-- Basic odd test, integer.
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_odd"(val int);
INSERT INTO "mediantest_odd" VALUES
    (3),
    (1),
    (3),
    (2);
-- ASCERTAIN
SELECT median(val) FROM mediantest_odd;
 median 
--------
      2
(1 row)

-------------------------------------------
-- Repeated values.
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_repeated"(val int);
INSERT INTO "mediantest_repeated"
SELECT 3 FROM generate_series(0, 1000) as T(i);
-- ASCERTAIN
SELECT median(val) FROM mediantest_repeated;
 median 
--------
      3
(1 row)

-------------------------------------------
-- Float test.
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_float"(val float);
INSERT INTO "mediantest_float" VALUES
    (3.0),
    (1.3),
    (90000.0),
    (-3),
    (15.123141),
    (-134141);
-- ASCERTAIN
SELECT median(val) FROM mediantest_float;
 median 
--------
    1.3
(1 row)

-------------------------------------------
-- Empty table.
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_empty"(val float);
-- ASCERTAIN
SELECT median(val) FROM mediantest_empty;
 median 
--------
       
(1 row)

-------------------------------------------
-- Gigantic odd table
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_gigantic_odd"(val int);
INSERT INTO mediantest_gigantic_odd (val)
SELECT i FROM generate_series(0, 100001) as T(i);
-- ASCERTAIN
SELECT median(val) FROM mediantest_gigantic_odd;
 median 
--------
  50000
(1 row)

-------------------------------------------
-- Gigantic even table
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_gigantic_even"(val int);
INSERT INTO mediantest_gigantic_even (val)
SELECT i FROM generate_series(0, 100000) as T(i);
-- ASCERTAIN
SELECT median(val) FROM mediantest_gigantic_even;
 median 
--------
  50000
(1 row)

-------------------------------------------
-- Multiple Selects
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_multiple"(a int, b int);
INSERT INTO mediantest_multiple (a, b)
SELECT i, 1000000 - i FROM generate_series(0, 10000) as T(i);
-- ASCERTAIN
SELECT median(a), median(b) FROM mediantest_multiple;
 median | median 
--------+--------
   5000 | 995000
(1 row)

-------------------------------------------
-- Strings
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_text"(val text);
INSERT INTO mediantest_text(val)
SELECT i::text FROM generate_series(0, 100000) as T(i);
-- ASCERTAIN
SELECT median(val) FROM mediantest_text;
 median 
--------
 54998
(1 row)

-------------------------------------------
-- Timestamp
-------------------------------------------
-- ARRANGE
CREATE TABLE "mediantest_date"(val timestamp);
INSERT INTO mediantest_date(val)
SELECT TIMESTAMP 'epoch' + (i * INTERVAL '1 second')
FROM generate_series(0, 100000) as T(i);
-- ASCERTAIN
SELECT median(val) FROM mediantest_date;
          median          
--------------------------
 Thu Jan 01 13:53:20 1970
(1 row)

